### PR TITLE
chore: add readme

### PR DESCRIPTION
**Pull request by JaeAeich**
This PR is for README.

This PR was branched on top of last PR #95 which had some changes:
- Pinned pip version because CI failed on resolving python deps and asked for a specific pip version range.
- docker-compose has been depricated and docker compose comes as plugin ever since v3.
- Added some type hints to resolve mypy errors.

## Summary by Sourcery

Documentation:
- Add a README file with information about the project, installation instructions, contribution guidelines, and contact information.

### CODE REVIEW

**Comment by JaeAeich** [comment]
@uniqueg Please ignore the commits from the previous PR, this PR is on top of it so the CI don't fail.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,180 @@
...
-See
-[ELIXIR Cloud outline repository](https://github.com/elixir-europe/elixir-cloud-outline)
-for details.
+## Synopsis
 
-More info coming soon.
+proWES is a robust and scalable [Global Alliance for Genomics and Health
+(GA4GH)][res-ga4gh] [Workflow Execution Service (WES) API][res-ga4gh-wes] gateway
+that may play a pivotal role in augmenting the capabilities of your GA4GH Cloud
+ecosystem by offering flexible middleware injection for effectively federating
```
Comment (uniqueg) [inline README.md]: > [...] for effectively federating [...]

From here on, this doesn't make sense - WES is not the same as TES. Maybe you could just say: 

> [...] for effective workflow deployment across a range of federated workflow management system deployments.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,180 @@
...
+## Description
+
+proWES gateway may serve as a crucial component in federated compute networks
+based on the GA4GH Cloud ecosystem. Its primary purpose is to provide
+centralized features to a federated network of independently operated GA4GH WES
+instances. As such, it can serve, for example, as a compatibility layer, a load
+balancer workload distribution layer, a public entry point to an enclave of
+independent compute nodes, or a means of collecting telemetry.
+
+When WES requests are received, proWES applies a configured middlewares before
```
Comment (uniqueg) [inline README.md]: Please replace with: 

> [...] applies one or more configurable middlewares [...]

**Review by uniqueg — CHANGES_REQUESTED** [review]

**Comment by uniqueg** [comment]
Maybe amend the description to say _why_ you pinned the Pip version. Also, you could mention that you added a type hint (presumably because Mypy complained), updated the Docker Compose call (as required by recent versions) and restricted the FOCA version.

**Comment by JaeAeich** [comment]
@sourcery-ai review

**Comment by JaeAeich** [comment]
> Maybe amend the description to say why you pinned the Pip version. Also, you could mention that you added a type hint (presumably because Mypy complained), updated the Docker Compose call (as required by recent versions) and restricted the FOCA version.

Those were for the last PR (those files aren't here anymore) but still added them here too.

**Review by uniqueg — APPROVED** [review]
Fixed a type, now LGTM :)

**Pull request merged**